Add a guide on prebuilds

bamurtaugh opened the pull request:
Fixes https://github.com/devcontainers/spec/issues/261. 

@craiglpeters I updated the info on the Kubernetes example a bit based on our chat last week - please feel free to propose other changes, or add in other tips/feedback from your experience working on the prebuild!

I also updated post.html to hopefully allow for 2 author posts correctly - cc @joshspicer in case you have any feedback on a different way to achieve this.

cc @olguzzar thanks again for the feedback - I'd love an additional review as I've incorporated your feedback and made a few more updates! I also created https://aka.ms/dev-container-prebuild that'll point to this guide for us to use in any docs or advocacy work.

Commit:
Scaffold post, enable multiple authors

Commit:
Add initial content

Commit:
Guide content updates

Commit:
Feedback

Commit:
Content and wording

samruddhikhandale reviewed (dismissed):
This is awesome and pretty helpful!
Thank you, Brigit and Craig 👏 ✨

joshspicer commented:
I opened this PR up in a codespace (with `bundle exec jeykll serve`) and i'm not seeing the authors anymore 🤔
<img width="1374" alt="image" src="https://github.com/devcontainers/devcontainers.github.io/assets/23246594/19dac477-e05c-4d8c-af8a-1b506efdc9ae">

joshspicer committed:
fix displaying author(s)

joshspicer commented:
Discussed my comment on authors with Brigit offline :)

olguzzar commented on _posts/2023-08-22-prebuild.md:
```diff
@@ -0,0 +1,134 @@
+* The main Kubernetes repo and any fork of it can now define a `.devcontainer` folder and [reference this prebuilt image](https://github.com/craiglpeters/kubernetes-devcontainer/blob/master/.devcontainer/devcontainer.json#L7) through: `"image": "ghcr.io/craiglpeters/kubernetes-devcontainer:latest"`
+
+**Dev container spec images**
+* This repo prebuilds a variety of dev containers, each of which is defined in their individual folks in the [src folder](https://github.com/devcontainers/images/tree/main/src)
```
should this say "in their individual **folders**"?

bamurtaugh replied:
It definitely should 😄 updating now

bamurtaugh replied:
Pushed a fix - thanks!

olguzzar commented:
This looks so great Brigit!!

Commit:
Typo

Commit:
Link between guides

Commit:
Feedback

bamurtaugh commented:
Incorporated feedback from yesterday's discussion (cross-link to this post from Chuck's guide on images vs dockerfile, added info on multi-stage builds). Would love another quick set of reviews, then I think we can merge this in and of course keep iterating in the future!

chrmarti commented on _posts/2022-12-16-dockerfile.md:
```diff
@@ -44,7 +46,7 @@ That's it! When you start up your Dev Container, the Dockerfile will be automati
 Better yet, you can can use a Dockerfile as a part of authoring an image you can share with others. You can even **add Dev Container settings and metadata right into the image itself**. This avoids having to duplicate config and settings in multiple devcontainer.json files and keeps them in sync with your images! 
 
-See the reference on **[pre-building](/implementors/reference/#prebuilding)** to learn more!
+See the guiide on **[pre-building](/_posts/2023-08-22-prebuild.md)** to learn more!
```
`guide`

bamurtaugh replied:
Thanks!

chrmarti reviewed (dismissed):
Looks great. Maybe a separate sample repo just showing off prebuilds would makes it easier for users to get an overview of what is needed. Not a requirement though.

bamurtaugh committed:
Typo

bamurtaugh commented:
Thanks so much @chrmarti! That's great feedback, and I've opened a tracking issue (in case anyone from the community happens to have one they'd like to contribute too!): https://github.com/devcontainers/devcontainers.github.io/issues/281.

markphip commented on _posts/2023-08-22-prebuild.md:
````diff
@@ -0,0 +1,138 @@
+     devcontainer build --workspace-folder . --push true --image-name <my_image_name>:<optional_image_version>
+     ```
+
+* You can automate pre-building your image by scheduling the build using a DevOps or continuous integration (CI) service like GitHub Actions. We've created a [GitHub Action](https://github.com/marketplace/actions/devcontainers-ci) and [Azure DevOps task](https://marketplace.visualstudio.com/items?itemName=devcontainers.ci) to help with this.
````
The link here to the GitHub Action does not work. Maybe it is not published in the Marketplace?

bamurtaugh replied:
Thanks for flagging - I'll merge this PR and then open a separate thread for that.
